plugins/browser-plugin-accelerator-ecommerce/src/api.ts

```diff
@@ -0,0 +1,228 @@
+import { BrowserPlugin, BrowserTracker, dispatchToTrackersInCollection } from '@snowplow/browser-tracker-core';
+import { SelfDescribingJson } from '@snowplow/tracker-core';
+import { buildEcommerceActionEvent } from './core';
+import {
+  CART_SCHEMA,
+  CHECKOUT_STEP_SCHEMA,
+  PAGE_SCHEMA,
+  PRODUCT_SCHEMA,
+  TRANSACTION_SCHEMA,
+  USER_SCHEMA,
+} from './schemata';
+import {
+  Cart,
+  CheckoutStep,
+  CommonEcommerceEventProperties,
+  ListClickEvent,
+  ListViewEvent,
+  Page as PageContext,
+  Product,
+  Transaction,
+  User as UserContext,
+} from './types';
+
+const _trackers: Record<string, BrowserTracker> = {};
+/* Scheduled to include Page and User */
+const _context: Record<string, SelfDescribingJson[]> = {};
+
+/**
+ * Adds ecommerce tracking
+ */
+export function EcommerceAcceleratorPlugin(): BrowserPlugin {
+  let trackerId: string;
+  return {
+    activateBrowserPlugin: (tracker) => {
+      trackerId = tracker.id;
+      _trackers[trackerId] = tracker;
+      _context[trackerId] = [];
+    },
+    contexts: () => {
+      return _context[trackerId] || [];
+    },
+  };
+}
+
+/**
+ * Track a checkout step
+ *
+ * @param event - The checkout step index and the attributes the user filled in.
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackCheckoutStep(
+  event: CheckoutStep & CommonEcommerceEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  const { context = [], timestamp, ...checkoutStep } = event;
+  context.push({ schema: CHECKOUT_STEP_SCHEMA, data: { ...checkoutStep } });
+
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildEcommerceActionEvent({ type: 'checkout_step' }), context, timestamp);
+  });
+}
+
+/**
+ * Track a product list view
+ *
+ * @param event - The list name along with the products which were viewed by the visitor.
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackProductListView(
+  event: ListViewEvent & CommonEcommerceEventProperties,
```

Comment:
Is the `ListViewEvent` really an event? Strictly speaking there is only one event type, the ecommerce action. Everything else seems like entities to me.

Also a nit: I wouldn't name the first argument `event` in these track functions but instead call them `listView` or `cart` or whatever they represent.